tests/petsc/assemble_matrix_parallel_01.cc

```diff
@@ -0,0 +1,493 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2009 - 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// same as deal.II/assemble_matrix_parallel_01, but for PETSc matrices
+// and vectors
+//
+// This test requires PETSc to be configured with the option
+// "--with-threadsafety" in case of debug builds of PETSc
+// For optimized builds, the above option is needed only in case
+// users want PETSc to produce useful logs with "-log_view" runtime
+// option
```

Comment:
We don't use `log_view` in tests but the first part could be a problem - we need to detect if users set up PETSc with a non-threaded configuration.

Reply:
do you mean in the test? I can add dummy prints in the test output in case `defined(PETSC_USE_DEBUG) && !defined(PETSC_HAVE_THREADSAFETY)` is true. Do you want me to reinstate the runtime error in the class constructor in such case?

Reply:
Can you clarify what you actually need? I believe that the test assembles the matrix in parallel, but does not actually create concurrent accesses to the matrix and vector. As a consequence, it should work with or without thread-safe PETSc builds. If that were true, you could just drop the paragraph in question here and we'd just have one more test.

Reply:
see my answer here https://github.com/dealii/dealii/pull/14716#discussion_r1087519083